flexx/app/_assetstore.py

```diff
@@ -433,16 +446,49 @@ def add_shared_data(self, name, data):
             raise TypeError('add_shared_data() name must be a str.')
         if name in self._data:
             raise ValueError('add_shared_data() got existing name %r.' % name)
-        if isinstance(data, str):
-            if data.startswith('file://'):
-                data = open(data.split('//', 1)[1], 'rb').read()
-            elif data.startswith(('http://', 'https://')):
-                data = urlopen(data, timeout=5.0).read()
-        if not isinstance(data, bytes):
-            raise TypeError('add_shared_data() data must be bytes.')
+        if isinstance(data, str) and data.startswith('file://'):
+            if not self.in_data_dir(data):
+                raise ValueError('add_shared_data() with a local file must exist '
+                                 'and be inside a data dir (see add_data_dir().')
+        elif isinstance(data, str) and data.startswith(('http://', 'https://')):
```

Comment:
That's what this does, except I called them data dirs and there can be multiple. I think we're good.